tools/gh_lists.py

```diff
@@ -50,7 +50,8 @@ def print_list(title, items):
             msg = "* `#{0} <{1}>`__: {2}"
             # sanitize whitespace, `, and *
             title = re.sub("\\s+", " ", issue.title.strip())
-            title = title.replace('`', '\\`').replace('*', '\\*')
+            title = re.sub("([^`])`([^`]|$)", "\g<1>``\g<2>", title)
```

Comment:
I get a doc build failure locally with this expression, via `python tools/gh_lists.py 1.15.0` to update the docs, with reason provided as:

```
/Users/treddy/github_projects/scipy_2/doc/source/release/1.15.0-notes.rst:142: WARNING: Inline literal start-string without end-string.
/Users/treddy/github_projects/scipy_2/doc/source/release/1.15.0-notes.rst:154: WARNING: Inline literal start-string without end-string.
```

The specific lines in question are:

```
* `#20773 <https://github.com/scipy/scipy/pull/20773>`__: BUG: sparse.csgraph, array types: support non-zero ``fill_value``s
* `#20855 <https://github.com/scipy/scipy/pull/20855>`__: TST: add additional margin to ``fail_slow``s
```

Looks like the issue is related to a backtick followed by a character at the end of a double-bactick section, when there is no intervening whitespace. If we add a space after the final backtick it looks a bit weird, but the doc build does pass.

With a manual hack to get around that for now, it does look nicer in general:

`main`:

![image](https://github.com/scipy/scipy/assets/7903078/d8447d8e-56b8-4e79-b757-7c4b4b442bff)


here:

![image](https://github.com/scipy/scipy/assets/7903078/ba70b46d-577d-4827-9840-141c8c8f681c)

Reply:
how about now?